server/src/main/scala/org/http4s/server/middleware/RequestLogger.scala

```diff
@@ -0,0 +1,29 @@
+package org.http4s
+package server
+package middleware
+
+import fs2._
+import org.log4s._
+import cats.implicits._
+import fs2.interop.cats._
+
+/**
+  * Simple Middleware for Logging Requests As They Are Processed
+  */
+object RequestLogger {
+  private[this] val logger = getLogger
+
+  def apply(service: HttpService)(implicit strategy: Strategy): HttpService = Service.lift{ req =>
+    Stream(req)
+      .observe(requestLogSink)
+      .evalMap[Task, Task, MaybeResponse]{req => service(req)}
+      .runFoldMap(identity)
+  }
+
+  def requestLogSink: Sink[Task, Request] = stream => {
+    stream
+      .flatMap(_.body)
+      .through(fs2.text.utf8Decode)
```

Comment:
What if the request has a declared charset of ISO-8859-1?  We could respect that.

What if it's binary?  We could sniff that out and render, oh, I don't know what.

What if it doesn't tell us what it is, but is not UTF-8?  If the sink fails, the stream fails.  Is there any sort of `attemptThrough`?  I don't think so.  Hmm.